urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/ui/TableroUI.java

```diff
@@ -0,0 +1,488 @@
+		mostrarTurno.setBackground(Color.WHITE);
+		mostrarTurno.setFont(estadisticasBlanco.getFont().deriveFont(Font.PLAIN));
+		mostrarTurno.setHorizontalAlignment(JLabel.CENTER);
+		
+		estadisticasNegro = new JLabel("Negras: ");
+		estadisticasNegro.setBorder(BorderFactory.createEtchedBorder());
+		estadisticasNegro.setBackground(Color.WHITE);
+		estadisticasNegro.setFont(estadisticasBlanco.getFont().deriveFont(Font.PLAIN));
+		estadisticasNegro.setHorizontalAlignment(JLabel.RIGHT);
+		
+		barraEstado = new JPanel(new GridLayout());
+		barraEstado.add(estadisticasBlanco);
+		barraEstado.add(mostrarTurno);
+		barraEstado.add(estadisticasNegro);
+		constrains.anchor = GridBagConstraints.PAGE_END;
+		constrains.fill = GridBagConstraints.HORIZONTAL;
+		constrains.weightx = 1;
+		constrains.gridx = 1;
+		constrains.gridy = 4;
+		pane.add(barraEstado, constrains);
+
+		boton = new JButton("Salir de Edición");
+		boton.setBorder(BorderFactory.createEtchedBorder());
+		boton.setSize(20, 50);
+		//boton.setBackground(new Color(0, 60, 0));
+		barraBoton = new JPanel(new GridLayout());
+		barraBoton.add(boton);
+		constrains = new GridBagConstraints();
+		constrains.fill = GridBagConstraints.NONE;
+		constrains.weightx = 300;
+		constrains.weighty = 100;
+		constrains.gridx = 1;
+		constrains.gridy = 20;
+		pane.add(barraBoton, constrains);
+		mostrarBarraEstado();
+		
+	}
+	
+	public void mostrarBarraEstado(){
+		barraBoton.setVisible(false);
+		barraEstado.setVisible(true);
+	}
+	
+	public void mostrarBarraBoton(){
+		barraEstado.setVisible(false);
+		barraBoton.setVisible(true);
+	}
+	
+
+	
+	public void definirCasilla(Point point, TipoCasilla tipoCasilla) {
+		ComponenteImagen imgcomp = FabricaImagenCasilla.construirCasilla(tipoCasilla);
+		int index = Transform.pointToIndex(point);
+		casillas.set(index, imgcomp);
+		tablero.remove(index);
+		tablero.add(imgcomp, index);
+		tablero.updateUI();
+	}
+	
+	public void marcarMovidasPosibles(Collection<Point> movidasPosibles, TipoCasilla color){
+		for (Point pssblPoint : movidasPosibles) {
+			definirCasilla(pssblPoint, color);
+			tablero.updateUI();
+		}
+		tablero.revalidate();
+	}
+	
+	public void desmarcarMovidasPosibles(Collection<Point> movidasPosibles){
+		for (Point pssblPoint : movidasPosibles) {
+			definirCasilla(pssblPoint, TipoCasilla.EMPTY);	
+			tablero.updateUI();
+		}
+		tablero.revalidate();
+	}
+	
+	public void rellenar(Collection<Point> puntosConRelleno, TipoCasilla color){
+		for (Point aRellenar : puntosConRelleno) {
+			definirCasilla(aRellenar, color);
+			tablero.updateUI();
+		}
+		tablero.revalidate();
+	}
+		
+	public void actualizarPuntaje(int estadisticasNegro, int estadisticasBlanco) {
+		this.estadisticasNegro.setText("Negras: " + estadisticasNegro);
+		this.estadisticasBlanco.setText("Blancas: " + estadisticasBlanco);
+	}
+	
+	public void actualizarTurno(String jugador){
+		jugador = jugador == "BLACK" ? "Negras" : "Blancas";
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.